src/esp/io/JsonEspTypes.h

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#pragma once
+
+/** @file
+ * @brief See JsonAllTypes.h. Don't include this header directly in user code.
+ */
+
+#include "JsonBuiltinTypes.h"
+#include "JsonMagnumTypes.h"
+
+#include "esp/assets/RenderAssetInstanceCreationInfo.h"
+#include "esp/core/esp.h"
+#include "esp/gfx/replay/Keyframe.h"
+
+namespace esp {
+namespace io {
+
+inline RJsonValue ToRJsonValue(const esp::vec3f& vec,
+                               RJsonAllocator& allocator) {
+  RJsonValue floatsArray(rapidjson::kArrayType);
+  for (int i = 0; i < vec.size(); i++) {
+    floatsArray.PushBack(vec.data()[i], allocator);
+  }
+  return floatsArray;
+}
+
+inline void FromRJsonValue(const RJsonValue& floatsArray, esp::vec3f& vec) {
+  // TODO: helper for objects that are simply arrays, with error-handling
+  ASSERT(floatsArray.Size() == vec.size());
+  for (int i = 0; i < vec.size(); i++) {
+    vec.data()[i] = floatsArray[i].GetFloat();
+  }
+}
```

Comment:
Many of the "from" functions already exist [here](https://github.com/facebookresearch/habitat-sim/blob/9c37189d0ac31d6cd5c918766d84b4a51128266e/src/esp/io/json.h#L54)